assets/src/vo/money.js

```diff
@@ -515,4 +516,45 @@ export default class Money {
 	static assertSameCurrency = ( currencyA, currencyB ) => {
 		assertSameCurrency( currencyA, currencyB );
 	}
+
+	/**
+	 * Receives an incoming value that could be a money formatted
+	 * string and returns a Money value object with the correct value
+	 * considering the provided currency.
+	 *
+	 * @param {string|number} moneyValue
+	 * @param {Currency} currency
+	 *
+	 * @return {Money} An instance of a money value object
+	 */
+	static fromMoneyValue = ( moneyValue, currency ) => {
+		assertCurrency( currency );
+		// detect if incoming value has a currency sign not matching provided
+		// currency.  This doesn't provide full protection from improper
+		// values sent in but is an initial safeguard.
```

Comment:
This check:

- throws an error if the incoming money value has more than one non numeric string (not counting commas or periods).
- throws an error if there is only one non numeric (not counting commas or periods) string that does not match the provided currency value object currency sign value.

Thus it's still possible to end up with an unexpected value if something like `10,10` is provided for a currency that uses `.` as the decimal place separator.  I think that would be an extreme edge case though so not worrying about it here.